crates/erg_linter/lint.rs

```diff
@@ -158,11 +158,30 @@ impl Linter {
             self.lint_too_many_params(chunk);
             self.lint_bool_comparison(chunk);
             self.lint_too_many_instance_attributes(chunk);
+            self.lint_tautology(chunk);
         }
         log!(info "Finished linting");
         self.warns.take()
     }
 
+    fn lint_tautology(&mut self, expr: &Expr) {
```

Comment:
Equality between expressions should be checked. Rules should also be applied recursively to the LHS and RHS.

```
assert x == y # OK
assert x == x # Warn
assert (x == x) == True # Warn
```